src/node.h

```diff
@@ -199,10 +199,18 @@ typedef intptr_t ssize_t;
 
 namespace node {
 
+class IsolateData;
+class Environment;
+
 // TODO(addaleax): Officially deprecate this and replace it with something
 // better suited for a public embedder API.
 NODE_EXTERN int Start(int argc, char* argv[]);
 
+// Teardown Node while it is running (active handles in the loop
+// and / or actively executing JS code). While callable from any
+// thread, make sure that the input environment is of the main thread.
```

Comment:
I would drop the last sentence and the check inside the function itself; this should work for any Environment, and we currently don’t expose a way to create non-main-thread Environments to embedders anyway (so they can’t really do anything besides call this for main threads)?

Reply:
thanks, made that amendments